bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/ScalarWebChannel.java

```diff
@@ -0,0 +1,351 @@
+     * Removes a property assignment
+     *
+     * @param key a non-null, non-empty key
+     * @return true if the property was found and removed, false otherwise
+     */
+    public @Nullable String removeProperty(final String key) {
+        Validate.notEmpty(key, "key cannot be empty");
+        return properties.remove(key);
+    }
+
+    /**
+     * Returns the properties for this channel based on the paths
+     *
+     * @return a non-null, non-empty properties
+     */
+    public Map<String, String> getProperties() {
+        final Map<String, String> props = new HashMap<>(properties);
+        props.put(ScalarWebChannel.CNL_PROPLEN, String.valueOf(paths.length));
+        for (int x = 0; x < paths.length; x++) {
+            props.put(ScalarWebChannel.CNL_PROPPREFIX + x, paths[x]);
+        }
+        props.put(ScalarWebChannel.CNL_CHANNELCATEGORY, category);
+        props.put(ScalarWebChannel.CNL_CHANNELID, id);
+        return props;
+    }
+
+    /**
+     * Gets the channel id based on group ID and channel id
+     *
+     * @return the channel id
+     */
+    public String getChannelId() {
+        return SonyUtil.createChannelId(service, createChannelId(category, id));
+    }
+
+    /**
+     * Helper method to create a simply channel id from the id (where the category
+     * is the id)
+     *
+     * @param id the non-null, non-empty id
+     * @return a non-null, non-empty channel id
+     */
+    public static String createChannelId(final String id) {
+        Validate.notEmpty(id, "id cannot be empty");
+        return createChannelId(id, id);
+    }
+
+    /**
+     * Helper method to create a channel id from the category/id
+     *
+     * @param category the non-null, non-empty category
+     * @param id the non-null, non-empty id
+     * @return a non-null, non-empty channel id
+     */
+    public static String createChannelId(final String category, final String id) {
+        Validate.notEmpty(id, "id cannot be empty");
+        return StringUtils.equalsIgnoreCase(category, id) ? id : (category + CATEGORYSEPARATOR + id);
+    }
+
+    @Override
+    public String toString() {
+        return getChannelId() + " (cid=" + id + ", ctgy=" + category + ", path=" + StringUtils.join(paths, ',') + ")";
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + category.hashCode();
+        result = prime * result + id.hashCode();
+        return result;
+    }
+
+    @Override
+    public boolean equals(final @Nullable Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        if (!(obj instanceof ScalarWebChannel)) {
+            return false;
+        }
+
+        final ScalarWebChannel other = (ScalarWebChannel) obj;
+        return StringUtils.equalsIgnoreCase(category, other.category) && StringUtils.equalsIgnoreCase(id, other.id);
```

Comment:
I think this breaks the primal rule of `hashCode` and `equals`. Your `hashCode` method above does not take into account your `equalsIgnoreCase` equality checking. You are allowing two objects to be equal but have different hashCodes. Fix!

Reply:
Good catch - fixed

Reply:
note: fixed in the other places as well